### PR TITLE
spindle: add img

### PR DESCRIPTION
**Pull request by tharvik**
Quite a simple figure, but does the trick of explaining the behavior of spindle (explained in text).

### CODE REVIEW

File: resources/incubator/images/spindle/network.svg [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+         sodipodi:role="line"
+         id="tspan39812"
+         x="85.337532"
+         y="170.93884"
+         style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-size:3.52778px;font-family:sans-serif;-inkscape-font-specification:'sans-serif, Normal';font-variant-ligatures:normal;font-variant-caps:normal;font-variant-numeric:normal;font-variant-east-asian:normal;stroke-width:0.264583">every network round, merge</tspan><tspan
+         sodipodi:role="line"
+         x="85.337532"
+         y="175.35387"
+         style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-size:3.52778px;font-family:sans-serif;-inkscape-font-specification:'sans-serif, Normal';font-variant-ligatures:normal;font-variant-caps:normal;font-variant-numeric:normal;font-variant-east-asian:normal;stroke-width:0.264583"
+         id="tspan39816">the hospital local's models</tspan></text>
```
Comment (cgrigis) [inline resources/incubator/images/spindle/network.svg:579]:
```suggestion
         id="tspan39816">the hospital local models</tspan></text>
```
It is not clear to me where this happens. Do all the hospitals, at every network round, merge the local models of all the other hospitals into their own local model?

File: resources/incubator/images/spindle/network.svg [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+         sodipodi:role="line"
+         x="111.56572"
+         y="70.506142"
+         style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-size:3.52777px;font-family:sans-serif;-inkscape-font-specification:'sans-serif, Normal';font-variant-ligatures:normal;font-variant-caps:normal;font-variant-numeric:normal;font-variant-east-asian:normal;stroke-width:0.264999;stroke-miterlimit:4;stroke-dasharray:none"
+         id="tspan38881-2">when model is trained,</tspan><tspan
+         sodipodi:role="line"
+         x="111.56572"
+         y="74.921173"
+         style="font-style:normal;font-variant:normal;font-weight:normal;font-stretch:normal;font-size:3.52777px;font-family:sans-serif;-inkscape-font-specification:'sans-serif, Normal';font-variant-ligatures:normal;font-variant-caps:normal;font-variant-numeric:normal;font-variant-east-asian:normal;stroke-width:0.264999;stroke-miterlimit:4;stroke-dasharray:none"
+         id="tspan39884">get prediction</tspan></text>
```
Comment (cgrigis) [inline resources/incubator/images/spindle/network.svg:552]: Suggestion: I would separate this in two parts:
1) The querier asks a node to train the model, which results in a globally trained model.
2) The querier sends data to a node and receives a prediction as the answer.

Something like:
```
--------------------------------------------------------------------------------------------------
|                                             Querier                                            |
--------------------------------------------------------------------------------------------------
                |  ↑                                                     |  ↑
    train model |  | done (model is trained)                       query |  | prediction
                ↓  |                                                     ↓  |
--------------------------------------------------------------------------------------------------
|                                            Hospital                                            |
--------------------------------------------------------------------------------------------------
```

File: views/incubator/spindle.tpl [hunk 3]
```diff
@@ -200,3 +200,27 @@
   won't be explained here. Let's say that the current state-of-the-art
   homomorphic encryption systems support both addition and multiplication.
 </p>
+
+<h2>SPINDLE</h2>
+
+<p>
+  SPINDLE uses the previous concepts, it allows to do some machine learning,
```
Comment (cgrigis) [inline views/incubator/spindle.tpl:207]:
```suggestion
  SPINDLE uses the previous concepts: it allows to do some machine learning,
```

File: views/incubator/spindle.tpl [hunk 4]
```diff
@@ -200,3 +200,27 @@
...
+  using homomorphic encryption to both ensure the privacy of data and to make
+  the machine learning computations. In the demonstrator, we present an use-case
```
Comment (cgrigis) [inline views/incubator/spindle.tpl:209]:
```suggestion
  the machine learning computations. In the demonstrator, we present a use-case
```

File: views/incubator/spindle.tpl [hunk 5]
```diff
@@ -200,3 +200,27 @@
...
+  based around predicting if a hospital patient is likely to have diabetes,
```
Comment (cgrigis) [inline views/incubator/spindle.tpl:210]:
```suggestion
  based around predicting whether a hospital patient is likely to have diabetes,
```

File: views/incubator/spindle.tpl [hunk 6]
```diff
@@ -200,3 +200,27 @@
...
+  based on the data of three different hospitals.
+</p>
+
+<p>
+  Machine learning usually never finishes, it does its computation many times,
+  with each round using the model created during the previous one. In the
```
Comment (cgrigis) [inline views/incubator/spindle.tpl:216]:
```suggestion
  with each round using the model created during the previous one, and hopefully improving it. In the
```

File: views/incubator/spindle.tpl [hunk 7]
```diff
@@ -200,3 +200,27 @@
...
+  demonstrator, you can set the number of wanted round. As to improve the
```
Comment (cgrigis) [inline views/incubator/spindle.tpl:217]:
```suggestion
  demonstrator, you can set the number of wanted round. In order to improve the
```

File: views/incubator/spindle.tpl [hunk 8]
```diff
@@ -200,3 +200,27 @@
...
+<p>
+  SPINDLE uses the previous concepts, it allows to do some machine learning,
+  using homomorphic encryption to both ensure the privacy of data and to make
+  the machine learning computations. In the demonstrator, we present an use-case
+  based around predicting if a hospital patient is likely to have diabetes,
+  based on the data of three different hospitals.
+</p>
+
+<p>
+  Machine learning usually never finishes, it does its computation many times,
```
Comment (cgrigis) [inline views/incubator/spindle.tpl:215]:
```suggestion
  The training phase in machine learning usually never finishes, it does its computation repeatedly,
```

File: views/incubator/spindle.tpl [hunk 9]
```diff
@@ -200,3 +200,27 @@
...
+  with each round using the model created during the previous one. In the
+  demonstrator, you can set the number of wanted round. As to improve the
+  overall speed of the algorithm, each hospital first computes a local model.
```
Comment (cgrigis) [inline views/incubator/spindle.tpl:218]:
```suggestion
  execution speed of the algorithm, each hospital first computes a local model.
```

**Review by cgrigis — COMMENTED** [review]
Some suggestions for clarity, feel free to apply or not. :)

File: resources/incubator/images/spindle/network.svg (hunk 2 above)
Comment (tharvik) [reply resources/incubator/images/spindle/network.svg:552]: Good point, I tried to simplify it by merging the two calls, but that doesn't helps with the following demo. I've splitted it.

**Pull request merged**